Source/utils/format.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "../player.h"
+
+#include <fmt/format.h>
+
+namespace fmt {
+
+template <>
+struct formatter<devilution::PLR_MODE> : formatter<string_view> {
+	template <typename FormatContext>
+	auto format(devilution::PLR_MODE mode, FormatContext &ctx)
+	{
+		string_view name = "unknown";
+		switch (mode) {
+		case devilution::PM_STAND:
```

Comment:
If you move this to a `static std::string_view GetPlrModeName(devilution::PLR_MODE mode)` the `switch` can be simplified:

```c++
static std::string_view GetPlrModeName(devilution::PLR_MODE mode) {
  switch (mode) {
    case devilution::PM_STAND: return "stand";
    case devilution::PM_WALK: return "walk (N, NW, or NE)";
    ...
    default: return "unknown";
  }
}
```

Reply:
Indeed. I this should go next to the enum declaration as it can be used in other places. That should also be `contrexpr`. Not sure about the name though. Maybe migrate the enums to `enum class` and then have one overloaded function called `toString`.

Reply:
I've created https://github.com/diasurgical/devilutionX/pull/1674 for this, but since `string_view` is missing from the Amiga cosstools I will merge that one after this PR. This PR will only include `fmt` and the logging functions update.